### PR TITLE
Use fixed number of iterations in sync-webgl-specific.

### PR DESCRIPTION
**Pull request by kenrussell**
Like the conformance2/query/occlusion-query test, use a fixed number
of iterations, as on slow automated test machines, a fixed duration
based on Date.now() leads to test flakiness.

Improve the synchronization with the GL server by using readPixels
instead of flush or finish.

### CODE REVIEW

**Comment by kenrussell** [comment]
@jdashg @kainino0x @jdarpinian @zhenyao please review.

This test has been flaky on our test bots since it was introduced, though it always passes locally (on much faster hardware). See http://crbug.com/818336 .

This changes the form of the test to be more like the preexisting occlusion-query test, which is tick-based rather than time-based.

File: sdk/tests/conformance2/sync/sync-webgl-specific.html [hunk 1]
```diff
@@ -91,28 +97,23 @@
...
 
-var startTimeOfFinish = 0;
 
 function finishSyncParameterTest() {
-  if (startTimeOfFinish == 0) {
-    startTimeOfFinish = Date.now();
-  }
-  if (Date.now() - startTimeOfFinish > signaling_retry_ms) {
+  if (--iteration == 0) {
     testFailed("Sync object wasn't signaled in a reasonable timeframe (" + (Date.now() - startTimeOfFinish) + " ms)");
```
Comment (zhenyao) [inline sdk/tests/conformance2/sync/sync-webgl-specific.html]: This startTimeOfFinish is no longer defined

Comment (kenrussell) [reply sdk/tests/conformance2/sync/sync-webgl-specific.html]: Thanks for catching this. Fixed.

**Comment by kenrussell** [comment]
Merging myself to try to get our bots reliably green again.

**Pull request merged**